backend/models/teamspaceSetting.js

```diff
@@ -129,37 +129,23 @@ class TeamspaceSettings {
 		labelFields.forEach((key) => {
 			if (utils.hasField(data, key)) {
 				if (Object.prototype.toString.call(data[key]) === "[object Array]") {
-					// store as key/val array
-					const labelObject = {};
-					data[key].forEach(label => {
-						if (label &&
-							Object.prototype.toString.call(label) === "[object String]" &&
-							label.trim()) {
-							// generate value from label
-							const value = label.trim().toLowerCase().replace(/ /g, "_").replace(/&/g, "");
-							if (labelObject[value]) {
-								switch (key) {
-									case "riskCategories":
-										throw responseCodes.RISK_DUPLICATE_CATEGORY;
-									case "topicTypes":
-										throw responseCodes.ISSUE_DUPLICATE_TOPIC_TYPE;
-								}
-							} else {
-								labelObject[value] = {
-									value,
-									label: label.trim()
-								};
+					const arrayUpdated = [];
+					const foundKeys = {};
+					for (let i = 0; i < data[key].length; ++i) {
+						const entry = data[key][i];
+						if(utils.isString(entry) && entry !== "") {
+							const value = entry.trim();
+							if (utils.hasField(foundKeys, value.toLowerCase())) {
+								throw responseCodes.DUPLICATED_ENTRIES;
```

Comment:
It seems like this error isn't thrown e.g. if topicTypes "Unknown" and "unknown" exist.

I also get this logged:

```
(node:6028) UnhandledPromiseRejectionWarning: TypeError: Cannot read property 'name' of undefined
    at Utils.mongoErrorToResCode (/Users/charencewong/Workspace/3drepo.io/backend/utils.js:146:10)
    at Object.responseCodes.respond (/Users/charencewong/Workspace/3drepo.io/backend/response_codes.js:397:19)
    at TeamspaceSettings.update.then.catch.err (/Users/charencewong/Workspace/3drepo.io/backend/routes/teamspace.js:523:18)
    at process._tickCallback (internal/process/next_tick.js:68:7)
(node:6028) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
```

Reply:
fixed